From 235a65786b205dc4988ee864dd3114fcb5e8cb68 Mon Sep 17 00:00:00 2001
From: Mitch Talmadge <mitcht@aptitekk.com>
Date: Wed, 20 Aug 2014 02:27:52 -0600
Subject: [PATCH 092/101] Added WIP Authenticator (currently it doesn't check
 whether or not the product has been bought. It is also missing the login
 dialog.) Updated gitignore.

---
 TeamScreen/.gitignore                         |   1 +
 .../TeamScreen/AutoUpdater/Authenticator.java | 106 ++++++++++++++++++
 .../TeamScreen/AutoUpdater/AutoUpdater.java   |  53 +++++----
 .../aptitekk/TeamScreen/TeamScreenDaemon.java |   7 ++
 4 files changed, 143 insertions(+), 24 deletions(-)
 create mode 100644 TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/Authenticator.java

diff --git a/TeamScreen/.gitignore b/TeamScreen/.gitignore
index 5c45529..3bcd007 100644
--- a/TeamScreen/.gitignore
+++ b/TeamScreen/.gitignore
@@ -1,5 +1,6 @@
 /bin
 TeamScreen.properties
 updates.properties
+auth.properties
 AutoUpdater.jar
 /updates
\ No newline at end of file
diff --git a/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/Authenticator.java b/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/Authenticator.java
new file mode 100644
index 0000000..1f3faeb
--- /dev/null
+++ b/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/Authenticator.java
@@ -0,0 +1,106 @@
+package com.aptitekk.TeamScreen.AutoUpdater;
+
+import java.awt.Image;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+
+public class Authenticator
+{
+    private static final Object synchronizedObject = new Object();
+    private Handshake handShake;
+    public static Image windowIcon;
+    private String userName;
+    private String passWord;
+    
+    public Authenticator(Image windowIcon)
+    {
+	Authenticator.windowIcon = windowIcon;
+	System.setProperty("javax.net.ssl.trustStore",AutoUpdater.getAbsDir()+"/secure/cacerts");
+	System.setProperty("javax.net.ssl.trustStorePassword","CpAhLLp94j");
+	this.loadProperties();
+    }
+    
+    private void loadProperties()
+    {
+	File propFile = new File(AutoUpdater.getAbsDir()+"/auth.properties");
+	try
+	{
+	    EncryptedProperties prop = new EncryptedProperties("&rA@l_43F^$!cD");
+	    if(!propFile.exists())
+	    {
+		FileOutputStream out = new FileOutputStream(propFile);
+		prop.setProperty("userName", "");
+		prop.setProperty("passWord", "");
+		this.userName = "";
+		this.passWord = "";
+		prop.store(out, "Authentication Properties File -- DO NOT EDIT.");
+		out.close();
+	    }
+	    else 
+	    {
+		FileInputStream in = new FileInputStream(propFile);
+		prop.load(in);
+		this.userName = prop.getProperty("userName");
+		this.passWord = prop.getProperty("passWord");
+		in.close();
+	    }
+	    this.userName = "bmineer";
+	    this.passWord = "admin10";
+	}
+	catch(FileNotFoundException e)
+	{
+	    e.printStackTrace();
+	}
+	catch(Exception e)
+	{
+	    e.printStackTrace();
+	}
+    }
+    
+    public boolean doAuthentication()
+    {
+	System.out.println("Authenticating...");
+	if(userName.isEmpty() || passWord.isEmpty())
+	{
+	    System.out.println("Empty credentials. Asking for user input.");
+	    return askForAuth();
+	}
+	else 
+	{
+	    System.out.println("Checking credentials with Server.");
+	    String result = checkAuthWithServer();
+	    if(result.equals("SUCCESS"))
+	    {
+		System.out.println("Authentication was Successful.");
+		return true;
+	    }
+	    else 
+	    {
+		System.out.println("Authentication Failed. Asking for user input.\nError was: "+result);
+		return askForAuth();
+	    }
+	}
+    }
+    
+    private boolean askForAuth()
+    {
+	
+	return false;
+    }
+    
+    private String checkAuthWithServer()
+    {
+	if(!AutoUpdater.generateHandshake())
+	    return "ERROR:handShake";
+	
+	String POSTData = "userName=" + userName + "&passWord="
+		+ passWord + "&handShake=" + AutoUpdater.handShake.getHandShakeResult();
+	
+	String result = AutoUpdater.sendPOSTGetString(AutoUpdater.updaterURLPrefix
+		+ AutoUpdater.updaterAuthenticatorSuffix, POSTData);
+	
+	return result;
+    }
+}
diff --git a/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdater.java b/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdater.java
index b2cd316..8a257eb 100644
--- a/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdater.java
+++ b/TeamScreen/src/com/aptitekk/TeamScreen/AutoUpdater/AutoUpdater.java
@@ -25,15 +25,16 @@ import org.apache.commons.io.FileUtils;
  */
 public class AutoUpdater
 {
-    /** Static variables that connection to the server requires.
+    /** 
+     * Static variables that connection to the server requires.
      */
     public static final String applicationName = "TeamScreen";
-    
-    /** Static variables that connection to the server requires.
+    /** 
+     * Static variables that connection to the server requires.
      */
     public static final String updaterURLPrefix = "https://AptiTekk.com/AutoUpdater/V1/";
-    /** Static variables that connection to the server requires.
-     * 
+    /** 
+     * Static variables that connection to the server requires.
      */
     public static final String updaterChangeLogSuffix = "ChangeLogFactory.php";
     /**
@@ -48,12 +49,16 @@ public class AutoUpdater
      * Static variables that connection to the server requires.
      */
     public static final String updaterHandshakeSuffix = "Handshake.php";
+    /** 
+     * Static variables that connection to the server requires.
+     */
+    public static final String updaterAuthenticatorSuffix = "Authenticator.php";
     
-    private int currentVersionID;
-    private String currentVersionName = "0.0_0.0";
+    private static int currentVersionID;
+    private static String currentVersionName = "0.0_0.0";
     private AutoUpdaterDownloadThread autoUpdaterDownloadThread;
     private static final Object synchronizedObject = new Object();
-    private Handshake handShake;
+    static Handshake handShake;
     private boolean automaticallyUpdate;
     public static Image windowIcon;
     private static ArrayList<UpdaterListener> updateListeners = new ArrayList<UpdaterListener>();
@@ -84,8 +89,8 @@ public class AutoUpdater
 		FileOutputStream out = new FileOutputStream(propFile);
 		prop.setProperty("currentVersionID", "0");
 		prop.setProperty("currentVersionName", "0.0_0.0");
-		this.currentVersionID = 0;
-		this.currentVersionName = "0.0_0.0";
+		currentVersionID = 0;
+		currentVersionName = "0.0_0.0";
 		prop.store(out, "Auto Updater Properties File -- DO NOT EDIT.");
 		out.close();
 	    }
@@ -93,8 +98,8 @@ public class AutoUpdater
 	    {
 		FileInputStream in = new FileInputStream(propFile);
 		prop.load(in);
-		this.currentVersionID = Integer.parseInt(prop.getProperty("currentVersionID"));
-		this.currentVersionName = prop.getProperty("currentVersionName");
+		currentVersionID = Integer.parseInt(prop.getProperty("currentVersionID"));
+		currentVersionName = prop.getProperty("currentVersionName");
 		in.close();
 	    }
 	}
@@ -118,8 +123,8 @@ public class AutoUpdater
 	String POSTData = "currentVersionID=" + currentVersionID + "&modType="
 		+ AutoUpdater.applicationName;
 	
-	String result = this.sendPOSTGetString(AutoUpdater.updaterURLPrefix
-		+ AutoUpdater.updaterVersionFinderSuffix, POSTData);
+	String result = sendPOSTGetString(updaterURLPrefix
+		+ updaterVersionFinderSuffix, POSTData);
 	if(result == null || result.startsWith("ERROR"))
 	{
 	    System.out.println("Error message received: " + result);
@@ -129,9 +134,9 @@ public class AutoUpdater
 	{
 	    int amountToUpdate = Integer.parseInt(result.split(":")[1]);
 	    System.out.println("Amount to Update: " + amountToUpdate);
-	    String changeLog = this.sendPOSTGetString(
-		    AutoUpdater.updaterURLPrefix
-		    + AutoUpdater.updaterChangeLogSuffix, POSTData);
+	    String changeLog = sendPOSTGetString(
+		    updaterURLPrefix
+		    + updaterChangeLogSuffix, POSTData);
 	    if(automaticallyUpdate || this.askUserToUpdate(changeLog, amountToUpdate))
 	    {
 		System.out.println("AutoUpdater granted permission to update. Generating Handshake...");
@@ -167,11 +172,11 @@ public class AutoUpdater
 	}
     }
     
-    private boolean generateHandshake()
+    static boolean generateHandshake()
     {
 	String POSTData = "currentVersionID=" + currentVersionID + "&modType="
 		+ AutoUpdater.applicationName;
-	String data = this.sendPOSTGetString(AutoUpdater.updaterURLPrefix
+	String data = sendPOSTGetString(AutoUpdater.updaterURLPrefix
 		+ AutoUpdater.updaterHandshakeSuffix, POSTData);
 	if(data != null && !data.isEmpty())
 	{
@@ -180,7 +185,7 @@ public class AutoUpdater
 		String[] split = data.split(":");
 		if(split.length == 3)
 		{
-		    this.handShake = new Handshake(split[1], split[2]);
+		    handShake = new Handshake(split[1], split[2]);
 		    return true;
 		}
 	    }
@@ -205,11 +210,11 @@ public class AutoUpdater
 	logMessage("Downloading Updates...", LevelEnum.INFO);
 	
 	this.autoUpdaterDownloadThread = new AutoUpdaterDownloadThread(
-		this.currentVersionID, amountToUpdate, this.handShake, automaticallyUpdate);
+		currentVersionID, amountToUpdate, handShake, automaticallyUpdate);
 	this.autoUpdaterDownloadThread.start();
     }
     
-    private String sendPOSTGetString(String URLString, String POSTData)
+    static String sendPOSTGetString(String URLString, String POSTData)
     {
 	URLConnection connection;
 	URL url;
@@ -267,7 +272,7 @@ public class AutoUpdater
      */
     public String getCurrentVersionName()
     {
-	return this.currentVersionName;
+	return currentVersionName;
     }
     
     /**
@@ -307,7 +312,7 @@ public class AutoUpdater
 	{
 	    if(listener instanceof UpdaterListener)
 	    {
-		this.updateListeners.add(listener);
+		updateListeners.add(listener);
 	    }
 	}
     }
diff --git a/TeamScreen/src/com/aptitekk/TeamScreen/TeamScreenDaemon.java b/TeamScreen/src/com/aptitekk/TeamScreen/TeamScreenDaemon.java
index 3398c6c..1570354 100644
--- a/TeamScreen/src/com/aptitekk/TeamScreen/TeamScreenDaemon.java
+++ b/TeamScreen/src/com/aptitekk/TeamScreen/TeamScreenDaemon.java
@@ -34,6 +34,7 @@ import javax.swing.SwingUtilities;
 import javax.swing.UIManager;
 import javax.swing.border.EmptyBorder;
 
+import com.aptitekk.TeamScreen.AutoUpdater.Authenticator;
 import com.aptitekk.TeamScreen.AutoUpdater.AutoUpdater;
 import com.aptitekk.TeamScreen.AutoUpdater.LevelEnum;
 import com.aptitekk.TeamScreen.AutoUpdater.UpdaterListener;
@@ -163,6 +164,12 @@ public class TeamScreenDaemon implements KeyListener, HotkeyListener, UpdaterLis
 
 			displayPopup(trayIcon, "Checking For Updates...", "TeamScreen is now checking for updates...", TrayIcon.MessageType.INFO);
 
+			if(!new Authenticator(FolderManager.createImage("/images/TeamScreen-64px.png", "tray icon")).doAuthentication())
+			{
+			    System.exit(0);
+			    return;
+			}
+			
 			AutoUpdater updater = new AutoUpdater(Boolean.valueOf(properties.getConfig().getProperty("AutoUpdate")), 
 					FolderManager.createImage("/images/TeamScreen-64px.png", "tray icon"));
 			updater.addUpdaterListener(this);
-- 
2.39.5 (Apple Git-154)

